### DISCUSSION TITLE
ConnectionPool.new(name: pool_name) - where is pool_name being used?

### DISCUSSION THREAD

**Discussion by Yurokle** [sidekiq/sidekiq#6445] [Q&A] [answered]
Hi. I'm going through Sidekiq 7 code and I saw this:
```
# redis_connection.rb
        # ...
        ConnectionPool.new(timeout: pool_timeout, size: size, name: pool_name) do
          redis_config.new_client
        end
```

the comment in `config.rb` code says: `# connection pool is lazy, it will not create connections unless you actually need them so don't be skimpy!`, I can see `pool_name` being passed to the `ConnectionPool` initializer, but I can't see it being used anywhere in the `connection_pool` gem. Am I missing something?

**Comment by mperham** [comment, marked as answer]
I think that was old code that never got implemented. Because sidekiq 7 added support for capsules it has to create multiple connection pools; I had thought about passing the name of the capsule to the pool to make debugging pool issues easier.

**Reply by Yurokle** [reply]
Gotcha, thanks Mike!
